File: Easy/struttura/meta_strutturaprincview/meta_strutturaprincview.cs

/*
Easy
Copyright (C) 2021 Università degli Studi di Catania (www.unict.it)
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/


using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_strutturaprincview
{
    public class Meta_strutturaprincview : Meta_easydata 
	{
        public Meta_strutturaprincview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "strutturaprincview") {
				Name = "Strutture";
			EditTypes.Add("princ");
			ListingTypes.Add("princ");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idstruttura"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "princ": {
						DescribeAColumn(T, "title", "Denominazione", nPos++);
						DescribeAColumn(T, "struttura_codice", "Codice", nPos++);
						DescribeAColumn(T, "struttura_codiceipa", "Codice IPA", nPos++);
						DescribeAColumn(T, "struttura_email", "E-Mail", nPos++);
						DescribeAColumn(T, "struttura_fax", "Fax", nPos++);
						DescribeAColumn(T, "aoo_title", "AOO", nPos++);
						DescribeAColumn(T, "sede_title", "Sede", nPos++);
						DescribeAColumn(T, "strutturakind_title", "Tipologia", nPos++);
						DescribeAColumn(T, "upb_title", "UPB", nPos++);
						DescribeAColumn(T, "struttura_telefono", "Telefono", nPos++);
						DescribeAColumn(T, "struttura_title_en", "Denominazione (ENG)", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "princ": {
						return "title asc ";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/struttura/meta_strutturaprincview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_strutturaprincview() {
        MetaData.apply(this, ["strutturaprincview"]);
        this.name = 'meta_strutturaprincview';
    }

    meta_strutturaprincview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_strutturaprincview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'princ':
						this.describeAColumn(table, 'title', 'Denominazione', null, 2000, 1024);
						this.describeAColumn(table, 'struttura_codice', 'Codice', null, 3000, 50);
						this.describeAColumn(table, 'struttura_codiceipa', 'Codice IPA', null, 4000, null);
						this.describeAColumn(table, 'struttura_email', 'E-Mail', null, 5000, 200);
						this.describeAColumn(table, 'struttura_fax', 'Fax', null, 6000, 50);
						this.describeAColumn(table, 'aoo_title', 'AOO', null, 7200, 1024);
						this.describeAColumn(table, 'sede_title', 'Sede', null, 8200, 1024);
						this.describeAColumn(table, 'strutturakind_title', 'Tipo', null, 9200, 50);
						this.describeAColumn(table, 'upb_title', 'UPB', null, 10200, 150);
						this.describeAColumn(table, 'struttura_telefono', 'Telefono', null, 11000, 50);
						this.describeAColumn(table, 'struttura_title_en', 'Denominazione (ENG)', null, 12000, 1024);
						this.describeAColumn(table, 'struttura_active', 'Attivo', null, 25000, null);
//$objCalcFieldConfig_princ$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idstruttura"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "princ": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('strutturaprincview', new meta_strutturaprincview('strutturaprincview'));

	}());
